scripts/build_plugin.py

```diff
@@ -0,0 +1,484 @@
+# Copyright (c) 2013 Shotgun Software Inc.
+#
+# CONFIDENTIAL AND PROPRIETARY
+#
+# This work is provided "AS IS" and subject to the Shotgun Pipeline Toolkit
+# Source Code License included in this distribution package. See LICENSE.
+# By accessing, using, copying or modifying this work you indicate your
+# agreement to the Shotgun Pipeline Toolkit Source Code License. All rights
+# not expressly granted therein are reserved by Shotgun Software Inc.
+
+"""
+Helper script to generate a toolkit plugin given a plugin source location.
+
+This will analyse the info.yml found in the plugin source location
+and create a plugin scaffold, complete with standard helpers and
+a primed bundle cache.
+"""
+
+# system imports
+from __future__ import with_statement
+import os
+import sys
+import socket
+import shutil
+import optparse
+import datetime
+import getpass
+
+# add sgtk API
+this_folder = os.path.abspath(os.path.dirname(__file__))
+python_folder = os.path.abspath(os.path.join(this_folder, "..", "python"))
+sys.path.append(python_folder)
+
+# sgtk imports
+from tank import LogManager
+from tank.util import filesystem
+from tank.errors import TankError
+from tank.platform import environment
+from tank.descriptor import Descriptor, descriptor_uri_to_dict, create_descriptor
+from tank.authentication import ShotgunAuthenticator
+from tank_vendor import yaml
+
+# set up logging
+logger = LogManager.get_logger("build_plugin")
+
+# required keys in the info.yml plugin manifest file
+REQUIRED_MANIFEST_PARAMETERS = ["base_configuration", "entry_point"]
+
+# the folder where all items will be cached
+BUNDLE_CACHE_ROOT_FOLDER_NAME = "bundle_cache"
+
+# when we are baking a config, use these settings
+BAKED_BUNDLE_NAME = "tk-config-plugin"
+BAKED_BUNDLE_VERSION = datetime.datetime.now().strftime("%Y%m%d_%H%M%S")
+
+# generation of the build syntax
+BUILD_GENERATION = 1
+
+class OptionParserLineBreakingEpilog(optparse.OptionParser):
+    """
+    Subclassed version of the option parser that doesn't
+    swallow white space in the epilog
+    """
+    def format_epilog(self, formatter):
+        return self.epilog
+
+def _cache_apps(sg_connection, cfg_descriptor, bundle_cache_root):
```

Comment:
This seems handy in general. Potentially make a method available in core? I see similar logic in `core_localize.py` too.